### ISSUE TITLE
mpv ignore correct "--screen-name=" option and playing video on default monitor.

### ISSUE THREAD

**Issue by IlyaIndigo** [mpv-player/mpv#9877]
### Reproduction steps
1 Set up via kscreen or otherwise the xinerama extension to extend the desktop to one or more connected monitors.
2 Run "xrandr --prop" to display the connected monitors and find out exactly what their "--screen-name=" is, which is the name and number of the video output and comes first before "connected".
3 Play the video on a monitor other than the default monitor using the command "mpv --no-config --screen-name=NAME video.mp4".
4 Play the video on a non-default monitor using "mpv --no-config --screen-name=1 video.mp4", going through the numbers starting with 1 if there are more than 2 monitors.

### Expected behavior
The video file should play on the specified monitor using both the "--screen-name=" option and the "--screen=" option.

### Actual behavior
The video file plays on the desired monitor only with the "--screen=" option, and the correct "--screen-name=" option is ignored and always plays the video file on the default monitor.

### Log file
[xrandr-prop.txt](https://github.com/mpv-player/mpv/files/8082007/xrandr-prop.txt)
```
#!/bin/bash
while read -r output hex conn; do
    [[ -z "$conn" ]] && conn=${output%%-*}
    echo "# $output $conn   $(xxd -r -p <<< "$hex")"
done < <(xrandr --prop | awk '
    !/^[ \t]/ {
        if (output && hex) print output, hex, conn
        output=$1
        hex=""
    }
    /ConnectorType:/ {conn=$2}
    /[:.]/ && h {
        sub(/.*000000fc00/, "", hex)
        hex = substr(hex, 0, 26) "0a"
        sub(/0a.*/, "", hex)
        h=0
    }
    h {sub(/[ \t]+/, ""); hex = hex $0}
    /EDID.*:/ {h=1}
    END {if (output && hex) print output, hex, conn}
    ' | sort
```
```
# DisplayPort-0 DisplayPort   DELL U2312HM
# DVI-0 DVI   VA249
# HDMI-0 HDMI   Panasonic-TV
```
`mpv --no-config --screen-name=DisplayPort-0 video.mp4`
```
 (+) Video --vid=1 (*) (h264 1280x720 29.970fps)
 (+) Audio --aid=1 --alang=eng (*) (aac 2ch 44100Hz)
AO: [alsa] 48000Hz stereo 2ch float
VO: [gpu] 1280x720 yuv420p
AV: 00:00:04 / 00:15:58 (0%) A-V:  0.000

Exiting... (Quit)
```

### Important Information
OpenSUSE Tumbleweed | RadeonSI (AMD Radeon 7770HD) | Xorg | KDE4
https://build.opensuse.org/package/show/multimedia:apps/mpv
`/usr/bin/mpv -v -v --version`
```
[cplayer] Command line options: '-v' '-v' '--version'
[cplayer] mpv 0.34.1+git.20220203T182214.5d31c7310c Copyright © 2000-2022 mpv/MPlayer/mplayer2 projects
[cplayer]  built on Feb  5 2022
[cplayer] FFmpeg library versions:
[cplayer]    libavutil       56.70.100
[cplayer]    libavcodec      58.134.100
[cplayer]    libavformat     58.76.100
[cplayer]    libswscale      5.9.100
[cplayer]    libavfilter     7.110.100
[cplayer]    libswresample   3.9.100
[cplayer] FFmpeg version: 4.4.1
[cplayer] 
[cplayer] Configuration: ./waf configure --prefix=/usr --bindir=/usr/bin --mandir=/usr/share/man --libdir=/usr/lib64 --docdir=/usr/share/doc/packages/mpv --confdir=/etc/mpv --enable-cdda --enable-dvdnav --enable-libmpv-shared --enable-manpage-build --enable-libarchive --enable-dvbin --enable-drm --enable-x11 --enable-wayland --enable-gl-wayland --enable-vulkan --enable-pipewire --enable-gl-x11 --enable-egl-x11 --enable-egl-drm --enable-zimg --disable-sdl2 --disable-build-date --disable-debug
[cplayer] List of enabled features: alsa asm caca cdda cplayer cplugins cuda-hwaccel cuda-interop drm dvbin dvdnav egl egl-drm egl-helpers egl-x11 ffmpeg ffnvcodec gbm gbm.h gl gl-wayland gl-x11 glibc-thread-name glob glob-posix gpl iconv jack javascript jpeg lcms2 libarchive libass libavdevice libbluray libdl libm libmpv-shared libplacebo libplacebo-next librt linux-fstatfs linux-input-event-codes lua luajit memfd_create optimize pipewire plain-gl posix posix-or-mingw pthreads pulse rubberband shaderc shaderc-shared stdatomic uchardet vaapi vaapi-drm vaapi-egl vaapi-vulkan vaapi-wayland vaapi-x-egl vaapi-x11 vapoursynth vdpau vdpau-gl-x11 vector vt.h vulkan wayland wayland-protocols x11 xv zimg zlib
[global] config path: '' -> '/home/ilya/.mpv'
[global] user path: '~~home/' -> '/home/ilya/.mpv'
[global] user path: '~~old_home/' -> ''
[global] config path: 'encoding-profiles.conf' -/-> '/home/ilya/.mpv/encoding-profiles.conf'
[global] config path: 'encoding-profiles.conf' -> '/etc/mpv/encoding-profiles.conf'
[cplayer] Reading config file /etc/mpv/encoding-profiles.conf
[cplayer] Applying profile 'default'...
[global] config path: 'mpv.conf' -/-> '/home/ilya/.mpv/mpv.conf'
[global] config path: 'config' -> '/home/ilya/.mpv/config'
[global] config path: 'mpv.conf' -> '/etc/mpv/mpv.conf'
[global] config path: 'config' -/-> '/etc/mpv/config'
[cplayer] Reading config file /etc/mpv/mpv.conf
[cplayer] Applying profile 'default'...
[cplayer] Reading config file /home/ilya/.mpv/config
[cplayer] Applying profile 'default'...
[cplayer] Setting option 'hwdec' = 'vaapi' (flags = 4)
[cplayer] Setting option 'fs' = 'yes' (flags = 4)
[cplayer] Setting option 'alang' = 'rus' (flags = 4)
[cplayer] Setting option 'slang' = 'forced' (flags = 4)
[cplayer] Setting option 'ytdl-format' = 'bestvideo[height<=720][fps<=?60][vcodec^=?avc]+bestaudio/best[height<=720][fps<=?60][vcodec^=?avc]' (flags = 4)
[cplayer] Setting option 'screenshot-directory' = '/tmp' (flags = 4)
[cplayer] Setting option 'v' = '' (flags = 8)
[cplayer] Setting option 'v' = '' (flags = 8)
[cplayer] Setting option 'version' = '' (flags = 8)
[cplayer] mpv 0.34.1+git.20220203T182214.5d31c7310c Copyright © 2000-2022 mpv/MPlayer/mplayer2 projects
[cplayer]  built on Feb  5 2022
[cplayer] FFmpeg library versions:
[cplayer]    libavutil       56.70.100
[cplayer]    libavcodec      58.134.100
[cplayer]    libavformat     58.76.100
[cplayer]    libswscale      5.9.100
[cplayer]    libavfilter     7.110.100
[cplayer]    libswresample   3.9.100
[cplayer] FFmpeg version: 4.4.1
[cplayer]
```

**Comment by Dudemanguy** [comment]
Most likely, the displayport screen is read in xrandr as screen 0 (when you use `--screen-name`), but in xinerama (what mpv uses to place the window on the selected screen) it's actually screen 1.

**Comment by IlyaIndigo** [comment]
> Most likely, the displayport screen is read in xrandr as screen 0 (when you use `--screen-name`), but in xinerama (what mpv uses to place the window on the selected screen) it's actually screen 1.

Let's assume this is the case, but I also have a TV on HDMI.
```
/usr/bin/mpv --no-config --screen-name=HDMI-0 video.mp4
 (+) Video --vid=1 (*) (h264 1280x720 29.970fps)
 (+) Audio --aid=1 --alang=eng (*) (aac 2ch 44100Hz)
AO: [alsa] 48000Hz stereo 2ch float
VO: [gpu] 1280x720 yuv420p
AV: 00:00:02 / 00:15:58 (0%) A-V:  0.000

Exiting... (Quit)
```
`/usr/bin/mpv --no-config --screen=2 video.mp4`

The behavior is similar.

Does it mean all 3 of my monitors are zeros?

**Comment by Dudemanguy** [comment]
Well no it shouldn't be possible for all three of them to be zeros. mpv iterates through all the different displays so the HDMI should be at least `1` or something. In theory it should have different behavior (different screen with a different `n`). Even if that `n` is wrong, it should be placed somewhere else. But I'm guessing you're saying it behaves exactly the same?

**Comment by IlyaIndigo** [comment]
> Even if that `n` is wrong, it should be placed somewhere else. But I'm guessing you're saying it behaves exactly the same?

Yes.
"--screen=" plays on the specified monitor.
"--screen-name=" always plays on the default Primary monitor and ignores this option.

**Comment by makigumo** [comment]
If I may chime in. I think I have a similar if not the same issue on macOS 12. I have a dual monitor setup.

* without "--screen" or "--screen-name" mpv opens on screen 1, where some versions ago it used to open on screen 0
* "--screen=0" opens on screen 0 as expected
* "--screen=0 -fs" opens on screen 1 
* "--fs-screen=0" opens on screen 1
* "--fs-screen=1" opens on screen 1 as expected

Version info
```
[cplayer] Command line options: '-v' '-v' '--version'
[cplayer] mpv 0.34.1 Copyright © 2000-2021 mpv/MPlayer/mplayer2 projects
[cplayer]  built on Thu Feb  3 02:39:50 UTC 2022
[cplayer] FFmpeg library versions:
[cplayer]    libavutil       56.70.100
[cplayer]    libavcodec      58.134.100
[cplayer]    libavformat     58.76.100
[cplayer]    libswscale      5.9.100
[cplayer]    libavfilter     7.110.100
[cplayer]    libswresample   3.9.100
[cplayer] FFmpeg version: 4.4.1
[cplayer] 
[cplayer] Configuration: waf configure --prefix=/usr/local/Cellar/mpv/0.34.1_1 --enable-html-build --enable-javascript --enable-libmpv-shared --enable-lua --enable-libarchive --enable-uchardet --confdir=/usr/local/etc/mpv --datadir=/usr/local/Cellar/mpv/0.34.1_1/share/mpv --mandir=/usr/local/Cellar/mpv/0.34.1_1/share/man --docdir=/usr/local/Cellar/mpv/0.34.1_1/share/doc/mpv --zshdir=/usr/local/Cellar/mpv/0.34.1_1/share/zsh/site-functions --lua=luajit
[cplayer] List of enabled features: asm bsd-fstatfs build-date cocoa coreaudio cplayer cplugins debug-build ffmpeg ffmpeg-aviocontext-bytes-read gl gl-cocoa glob glob-posix gpl iconv javascript jpeg lcms2 libarchive libass libavdevice libbluray libdl libm libmpv-shared lua luajit macos-10-11-features macos-10-12-2-features macos-10-14-features macos-cocoa-cb macos-media-player macos-touchbar optimize osx-thread-name plain-gl posix posix-or-mingw pthreads rubberband stdatomic swift uchardet vapoursynth vector videotoolbox-gl videotoolbox-hwaccel zimg zlib
[global] config path: '' -> '/Users/dan/.config/mpv'
[global] user path: '~~home/' -> '/Users/dan/.config/mpv'
[global] user path: '~~old_home/' -> '/Users/dan/.mpv'
[global] config path: 'encoding-profiles.conf' -/-> '/Users/dan/.config/mpv/encoding-profiles.conf'
[global] config path: 'encoding-profiles.conf' -/-> '/Users/dan/.mpv/encoding-profiles.conf'
[global] config path: 'encoding-profiles.conf' -> '/usr/local/etc/mpv/encoding-profiles.conf'
[cplayer] Reading config file /usr/local/etc/mpv/encoding-profiles.conf
[cplayer] Applying profile 'default'...
[global] config path: 'mpv.conf' -> '/Users/dan/.config/mpv/mpv.conf'
[global] config path: 'config' -/-> '/Users/dan/.config/mpv/config'
[global] config path: 'mpv.conf' -/-> '/Users/dan/.mpv/mpv.conf'
[global] config path: 'config' -/-> '/Users/dan/.mpv/config'
[global] config path: 'mpv.conf' -/-> '/usr/local/etc/mpv/mpv.conf'
[global] config path: 'config' -/-> '/usr/local/etc/mpv/config'
[cplayer] Reading config file /Users/dan/.config/mpv/mpv.conf
[cplayer] Applying profile 'default'...
[cplayer] Setting option 'hwdec' = 'auto' (flags = 4)
[cplayer] Setting option 'sub-font-size' = '44' (flags = 4)
[cplayer] Setting option 'slang' = 'jap,eng' (flags = 4)
[cplayer] Setting option 'alang' = 'jap,kor,eng' (flags = 4)
[cplayer] Setting option 'v' = '' (flags = 8)
[cplayer] Setting option 'v' = '' (flags = 8)
[cplayer] Setting option 'version' = '' (flags = 8)
[cplayer] mpv 0.34.1 Copyright © 2000-2021 mpv/MPlayer/mplayer2 projects
[cplayer]  built on Thu Feb  3 02:39:50 UTC 2022
[cplayer] FFmpeg library versions:
[cplayer]    libavutil       56.70.100
[cplayer]    libavcodec      58.134.100
[cplayer]    libavformat     58.76.100
[cplayer]    libswscale      5.9.100
[cplayer]    libavfilter     7.110.100
[cplayer]    libswresample   3.9.100
[cplayer] FFmpeg version: 4.4.1
[cplayer] 
```

**Comment by Dudemanguy** [comment]
macOS is completely unrelated. OP's issue has to do with the way mpv handles xinerama and xrandr (xorg).

**Comment by Dudemanguy** [comment]
@IlyaIndigo: Apologies for the huge delay on this. I got around to setting this one up and I see the issue. I'll put up a PR in a bit.

**Comment by IlyaIndigo** [comment]
@Dudemanguy Thank you!
If you need to test, please make a patch relative to the version used in openSUSE https://build.opensuse.org/package/show/multimedia:apps/mpv I will apply and test.

**Comment by Dudemanguy** [comment]
Fixed by [#10066](https://github.com/mpv-player/mpv/pull/10066).
